### PR TITLE
Ad738x update

### PR DESCRIPTION
**Pull request by amiclaus**
Update ad738x driver/project using the latest platform drivers.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/adc/ad738x/ad738x.c [hunk 1]
```diff
@@ -159,19 +159,13 @@ int32_t ad738x_spi_single_conversion(ad738x_dev *dev,
...
+				   enum ad738x_conv_mode mode)
 {
-	int32_t ret;
-
-	ret = ad738x_spi_write_mask(dev,
-				    AD738X_REG_CONFIG2,
-				    AD738X_CONFIG2_SDO2_MSK,
-				    AD738X_CONFIG2_SDO2(mode));
-
-	dev->conv_mode = mode;
```
Comment (scuciurean) [inline drivers/adc/ad738x/ad738x.c:172]: Why drop the conv_mode parameter and assignment? if the spi_write failed the conv_mode shouldn't be updated

Comment (amiclaus) [reply drivers/adc/ad738x/ad738x.c:172]: I prefered using a standard driver approach:

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L330

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L338

Comment (amiclaus) [reply drivers/adc/ad738x/ad738x.c:172]: I prefered using a standard driver approach:

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L330

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L338

**Review by mchindri — APPROVED** [review]
LGTM

**Review by amiclaus — COMMENTED** [review]
V2:
- fix header comment for `parameters.h`

**Pull request merged**